swaglyrics/__main__.py

```diff
@@ -29,19 +31,40 @@ def main():
 
 	parser.add_argument('-t', '--tab', action='store_true', help='Display lyrics in a browser tab.')
 	parser.add_argument('-c', '--cli', action='store_true', help='Display lyrics in the command-line.')
+	parser.add_argument('-cr', '--chrome', action='store_true', help='Display lyrics from browser.')
 
 	args = parser.parse_args()
+	app.template_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'templates')
+	app.static_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'static')
+	port = 5042  # random
+	url = "http://127.0.0.1:{port}".format(port=port)
+	
+	
+	#Function to check whether if chrome was passed via arguments or not
+	def modeChecker(argg):
+		chrome.initvariables()
+		if argg:
+			chrome.isChrome = True
+			print("Listesning to chrome")
+		else:
+			chrome.isChrome = False
+			print("Listesning to local")
+
 
 	if args.tab:
 		print('Firing up a browser tab!')
-		app.template_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'templates')
-		app.static_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'static')
-		port = 5042  # random
-		url = "http://127.0.0.1:{port}".format(port=port)
+		modeChecker(args.chrome)
 		threading.Timer(1.25, lambda: webbrowser.open(url)).start()
 		app.run(port=port)
 
 	elif args.cli:
+		modeChecker(args.chrome)
```

Comment:
Why do you need to start a server here? If `args.chrome` has been passed then `spotify.py` should feed the values from the extension so you that don't need to change cli and tab in two places.

Reply:
because if I want to get lyrics in terminal, there is no server instantiated. 

And if there is no server, Chrome extension cant post data to a location

Reply:
That should be happening in spotify.py so there is no redundancy. Look at my other comments 👍

Reply:
So you mean starting server from Spotify.py?
 
Yes, I tried. But it didn't work because it wasn't able to `import app from tab`.